### PR TITLE
planner: rename duplicated anonymous columns when creating view

### LINKED ISSUE #29326: CREATE View resulting "duplicate column name" error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
drop view if exists v1;
select 1, 1; # works
create view v1 as select 1, 1; # fails
select * from v1;
```
### 2. What did you expect to see? (Required)
```sql
mysql> drop view if exists v1;
Query OK, 0 rows affected (0.01 sec)

mysql> select 1, 1; # works
+---+---+
| 1 | 1 |
+---+---+
| 1 | 1 |
+---+---+
1 row in set (0.00 sec)

mysql> create view v1 as select 1, 1;
Query OK, 0 rows affected (0.04 sec)

mysql> select * from v1;
+---+------------+
| 1 | Name_exp_1 |
+---+------------+
| 1 |          1 |
+---+------------+
1 row in set (0.00 sec)
```
### 3. What did you see instead (Required)
```sql
mysql> drop view if exists v1;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> select 1, 1; # works
+---+---+
| 1 | 1 |
+---+---+
| 1 | 1 |
+---+---+
1 row in set (0.00 sec)

mysql> create view v1 as select 1, 1;
ERROR 1060 (42S21): Duplicate column name '1'
```
### 4. What is your TiDB version? (Required)

```sql
+-------------------------+--------------------------------------------------------------------------+
| Variable_name           | Value                                                                    |
+-------------------------+--------------------------------------------------------------------------+
| innodb_version          | 5.6.25                                                                   |
| protocol_version        | 10                                                                       |
| tidb_analyze_version    | 2                                                                        |
| tidb_row_format_version | 2                                                                        |
| tls_version             | TLSv1,TLSv1.1,TLSv1.2                                                    |
| version                 | 5.7.25-TiDB-v5.2.1                                                       |
| version_comment         | TiDB Server (Apache License 2.0) Community Edition, MySQL 5.7 compatible |
| version_compile_machine | x86_64                                                                   |
| version_compile_os      | osx10.8                                                                  |
+-------------------------+--------------------------------------------------------------------------+
```

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #29326

Problem Summary:

To make a compatibility with mysql, create view should not fail when the result contains some duplicated names. For example:

```
mysql> create view v as select 3, 4, 3, 4, 3, 4, 3, 4;
Query OK, 0 rows affected (0.01 sec)

mysql> select * from v \G
*************************** 1. row ***************************
           3: 3
           4: 4
  Name_exp_3: 3
  Name_exp_4: 4
Name_exp_1_3: 3
Name_exp_1_4: 4
Name_exp_2_3: 3
Name_exp_2_4: 4
1 row in set (0.00 sec)
```

By the way, a table column or a column name specified with `as` should not be renamed even if it is duplicated:

```
mysql> create table t(id int);
Query OK, 0 rows affected (0.03 sec)

mysql> create view v as select id, 1 as id from t;
ERROR 1060 (42S21): Duplicate column name 'id'
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Support rename duplicated column names when creating view
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]
MySQL really look cool(
BTW do we need to document this change?

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -3217,13 +3218,49 @@ func (b *PlanBuilder) addAliasName(ctx context.Context, selectFields []*ast.Sele
 		}
 	}
 
+	// dedupMap is used for renaming a duplicated anonymous column
+	dedupMap := make(map[string]int)
```
Comment (AilinKid) [inline planner/core/logical_plan_builder.go:3222]: s / make(map[string]int) / make(map[string]struct{})

**Comment by lcwangchao** [comment]
> MySQL really look cool( BTW do we need to document this change?

I'm not sure. We did not detail the behaviors for `CREATE VIEW ...` and just listed some compatibility descriptions. Maybe we need not to document this because it has the some behavior of mysql.

**Pull request merged**